packages/filter/src/query.ts

```diff
@@ -560,16 +562,21 @@ export class FilterBuilder<MT extends object = AnyObject> {
    * @param f - A field name to be included, an array of field names to be
    * included, or an Fields object for the inclusion/exclusion
    */
-  fields(...f: (Fields<MT> | (keyof MT)[] | keyof MT)[]): this {
+  fields(...f: (Fields<MT> | Extract<keyof MT, string>)[]): this {
```

Comment:
Do we still need to specify all forms? I think `Extract<...>` is already covered by `Fields<>` type now, isn't it? Would the following work?

```suggestion
  fields(...f: (Fields<MT>)[]): this {
```

Reply:
that cover calling it with 
```ts
builder.fields('property', 'property2')
```
while Fields<MT> only cover
```ts
builder.fields(['property', 'property2'])
// and
builder.fields({property: true, property2: true})
so i kept it in order to no introduce a breaking change

Reply:
I see, I did not realize `Extract<keyof MT, string>` is describing a single item (field name) only, not an array.

+1 to preserve backwards compatibility 👍🏻